### ISSUE TITLE
[feature] Option for uploading multiple specific references

### ISSUE THREAD

**Issue by Sebiee** [conan-io/conan#6925]
- [x] I've read the [CONTRIBUTING guide](https://github.com/conan-io/conan/blob/develop/.github/CONTRIBUTING.md).

I understand there is some work in progress (https://github.com/conan-io/conan/issues/5743) that might fulfill this use-case I'm gonna explain, but I just want to be sure it will really be possible when that issue is implemented, or maybe I can go other way to achieve what I want. Anyway, I'd be happy to have your input on this.

### Summary

I need the possibility to be more specific about the references I want to upload, i.e. I'd like to be able to do something like:
```bash
conan upload -r my_remote "boost/1.1.1@auth/ch" "zlib/1.2.3@auth/ch" "libressl/1.2.3@auth/ch" --all --parallel
```
Which will end up only uploading (compressing etc.) the packages of those references. Is there any way to achieve this atm?

### Motivation

In our company's ecosystem we applied an on-demand approach for building the conan packages for different environments (archs/flags/build_types etc.), instead of being required to build in advance packages for all our environments settings (which are many). We only upload to the remote conan server the recipe and a package with default settings, so the consumers will be able to build the desired package by their own.
Thus, when running the pipeline of a consumer app, we do the conan install with `--build missing`, building the dependent packages with the required settings. In order to avoid building always those packages, after the install command, we upload the packages we have in local cache.

Up until now, we used to always do `conan upload '*'` after `conan install --build missing` but we really want to get rid of this since we found out the upload command would take a lot of time (because it is compressing large deps like boost) even if no upload would need to be made (no package build in the `install` phase).
So we'd like to only upload the packages we built (we'll parse the json generated by the install command for this) and also upload them in parallel (so running multiple conan upload commands is not of so much help).

**Comment by jgsogo** [comment]
Hi, @Sebiee 

We are already redesigning the Conan CLI for the next major version and this is something to consider for the new set of commands and arguments (@czoido is leading that effort, he will consider this suggestion). For the current CLI I'm afraid that the best I can suggest you is to parse the JSON and run several `conan upload` commands 😞 

---

**Another approach (workaround) would be to build the libraries one by one and upload only the ones that have been created**. Instead of running `conan install --build missing`, you can create the lockfile of your consumer app (this lockfile contains all the dependencies, options, profiles,...), then run `conan graph build-order` to know the packages that need to be built, iterate this list of packages one by one running a `conan create` + `conan upload` using that lockfile (to guarantee it uses the same profile and options). This way, you would upload only the packages you have created.

This is the flow we recommend for a scalable CI, using the `build-order` Conan will tell you the packages that can be built in parallel, so you can split the work into several jobs and speed it up a little bit.

Let me know if this helps to alleviate your pain, and if you need help to making your mind up through the `conan graph` command.

Thanks!

**Comment by Sebiee** [comment]
Thank you @jgsogo!

The `conan graph` approach indeed seems helpful, but a little more complex. We're currently opened to small changes in our approach, so I don't think it will be implemented. Anyway it's good to know about this. Parallel build of missing dependencies sounds good.

For the moment, we're ok to wait and look forward for the next major version. Keep up the great work! 😊

**Comment by memsharded** [comment]
the ``package-list`` feature will be merged for 2.0.7 in https://github.com/conan-io/conan/pull/13928, that will implement this flow (and more):

```bash
conan create app --format=json > graph.json
conan list --graph=graph.json --format=json > pkglist.json
conan upload --list=pkglist.json -r=default
```
